dotCMS/src/main/java/com/liferay/portal/ejb/CompanyManagerUtil.java

```diff
@@ -196,12 +197,14 @@ public static void updateUsers(java.lang.String languageId,
 	throws PortalException, SystemException,com.dotmarketing.exception.DotRuntimeException {
 		try{		
 			CompanyManager companyManager = CompanyManagerFactory.getManager();
+			if (!DBTimeZoneCheck.isTimeZoneValid(timeZoneId)) {
+				throw new PortalException(String.format("Invalid Timezone %s", timeZoneId));

```

Comment:
I am not sure if a PortalException is the better option here, maybe something like a IllegalArgumentException could be better

Reply:
The problem is if it's not a `PortalException` or `SystemExecption` the upper calling resource class will wrap any other exception in a `SystemException` causing the error message displayed to the user to include the exception fully qualified name like this: `java.lan.IllegalArgumentException: Invalid timezone CST`.